sdk/src/main/java/com/bugsnag/android/ExceptionHandler.java

```diff
@@ -77,12 +77,12 @@ public void uncaughtException(@NonNull Thread thread, @NonNull Throwable throwab
                 StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.LAX);
 
                 client.cacheAndNotify(throwable, Severity.ERROR,
-                    metaData, severityReason, violationDesc);
+                    metaData, severityReason, violationDesc, thread);
```

Comment:
This bit needs adjustment; since the thread has crashed, it may get recycled in the pool to do something else, depending on which thread crashed. Other threads are in an approximate state relative to when the exception occurred given that some may have started or died in the interim, which is expected. 

However, all of the required elements are here to construct the expected result. For unhandled reports, the `errorReportingThread` should have:

1. `thread.getId()`
2. `thread.getName()`
3. `exception.getStackTrace()`

The easiest way to accomplish this may be to replace the stack for the error reporting thread to use the exception stack if the exception is unhandled.